src/Network/WebSockets/TSSocketManager.m

```diff
@@ -250,12 +266,22 @@ - (void)closeWebSocket
 
 - (void)webSocketDidOpen:(SRWebSocket *)webSocket {
     OWSAssert([NSThread isMainThread]);
-    
+    OWSAssert(webSocket);
+    if (webSocket != self.websocket) {
+        // Ignore events from obsolete web sockets.
+        return;
+    }
```

Comment:
This (and the corresponding changes below) does turn out to be necessary.  Apparently the websocket captures it's delegate _before_ dispatching the delegate method invocation.